x-pack/solutions/observability/plugins/investigate_app/scripts/evaluate/kibana_client.ts

```diff
@@ -0,0 +1,326 @@
+        that.getUrl({
+          pathname: '/api/observability/investigations',
+        }),
+        body
+      );
+      return response.data.id;
+    }
+
+    async function deleteInvestigation({
+      investigationId,
+    }: Parameters<RCAChatClient['deleteInvestigation']>[0]) {
+      await that.axios.delete(
+        that.getUrl({
+          pathname: `/api/observability/investigations/${investigationId}`,
+        })
+      );
+    }
+
+    async function rootCauseAnalysis({
+      connectorIdOverride,
+      investigationId,
+      from,
+      to,
+      alert,
+    }: {
+      connectorIdOverride?: string;
+      investigationId: string;
+      from: string;
+      to: string;
+      alert?: EcsFieldsResponse;
+    }) {
+      const chat$ = defer(() => {
+        that.log.debug(`Calling root cause analysis API`);
+        const serviceName = alert?.['service.name'] as string | undefined;
+        if (!alert) {
+          throw new Error(
+            'Alert not found. Please ensure you have loaded test fixture data prior to running tests.'
+          );
+        }
+        if (!serviceName) {
+          throw new Error(
+            'Service name is missing from the alert data. Please double check your alert fixture.'
+          );
+        }
+        const context = getRCAContext(alert, serviceName);
+        const body = {
+          investigationId,
+          connectorId: connectorIdOverride || '34002e09-65be-4332-9084-2649c86abacb',
+          context,
+          rangeFrom: from,
+          rangeTo: to,
+          serviceName: 'controller',
+          completeInBackground: false,
+        };
+
+        return that.axios.post(
+          that.getUrl({
+            pathname: '/internal/observability/investigation/root_cause_analysis',
+          }),
+          body,
+          { responseType: 'stream', timeout: NaN }
+        );
+      }).pipe(
+        switchMap((response) => streamIntoObservable(response.data)),
+        serializeRCAResponse(),
+        tap((event) => {
+          switch (true) {
+            case event.includes('"name":"endProcessAndWriteReport"'):
+              that.log.info(`Root cause analysis completed.`);
+              break;
+            case event.includes('"name":"observe"'):
+              that.log.info(`Observing...`);
+              break;
+            case event.includes('"name":"error"'):
+              that.log.error(`Encountered an error!`);
+              break;
+            case event.includes('"name":"investigateEntity"'):
+              that.log.info(`Investigating entity...`);
+              break;
+            default:
+              that.log.info(`Analyzing root cause...`);
+              break;
+          }
+        }),
+        toArray()
+      );
+
+      const message = await lastValueFrom(chat$);
+      const eventString = message.join('');
+
+      const events = extractEvents(eventString, []);
+
+      return events;
+    }
+
+    function extractEvents(
```

Comment:
you'll probably want to have a look at https://github.com/elastic/kibana/blob/0147e7456e4540dc691bdb10b46f58adaa1240b4/src/platform/packages/shared/kbn-sse-utils-client/index.ts#L10